Update CHANGELOG for 0.10 (#590)
IntelPython/dpctl: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.10.0] - 09/xx/2021

### Added

> - Used clang-format off, clang-format on to avoid include reordering (#588)

Reviewer: ```suggestion
- Used clang-format off, clang-format on to avoid include reordering in pybind11 example (#588)
```

> - Add methods __bool__, __float__ and __int__ to usm_ndarray (#578)

Reviewer: ```suggestion
- Added methods __bool__, __float__, __int__, __index__, and __complex__ to usm_ndarray (#578)
```

- Added test based on example from #583
- Added tests for constructors with invalid capsules (#577)

> - Coverage sycl queue (#574)

Reviewer: ```suggestion
- Improved coverage of sycl queue implementation (#574)
```

- Added a test to exercise API exported function (get_event_ref). (#570)
- Expanded tests in test_sycl_context to improve coverage (#571)
- Tweaks to test_sycl_event to improve coverage (#567)

> - Improve coverage of dpctl.__init__ file and other service functions (#563)

Reviewer: ```suggestion
- Improved coverage of dpctl.__init__ file and other service functions (#563)
```

- Added test for repr and test for default argument to constructor (#565)
- Added some tests to involve capsule (#564)

> - Add workflow for Public CI on Windows (#534)

Reviewer: ```suggestion
- Added workflow for Public CI on Windows (#534)
```

- DPCTLQueue_Memcpy, _Prefetch, _Memadvise become asynchronous (#557)

> - Add aspect selector (#558)

Reviewer: ```suggestion
- Add device aspect selector, `dpctl.select_device_with_aspects` (#558)
```

### Changed
- Parametrize ocl kernel compiled from source in types of arguments (#581)

> - Temporary off self-hosted runner (#559)

Reviewer: ```suggestion
- Temporary disabled self-hosted CI jobs runner (#559)
```

> - Static method SyclQueue._create_from_context_and_device change (#579)

Reviewer: ```suggestion
- Changed static method `SyclQueue._create_from_context_and_device` (#579)
```

> - Transition all Python API to use pytest over unittest, improved coverage in dpctl/memory (#575)

Reviewer: ```suggestion
- Transitioned all Python API to use pytest over unittest, improved coverage in dpctl/memory (#575)
```

> - profiling_info_submit was not property (#573)

Reviewer: ```suggestion
- Changed `dpctl.SyclEvent.profiling_info_submit` from method to a property (#573)
```

- Simplified arg parsing in SyclDevice constructor (#572)

> - Try using <img> tag with alignment attribute set (#562)

Reviewer: ```suggestion
- Used<img> tag with alignment attribute set in README (#562)
```

- Moved sycl timer into dpctl.SyclTimer (#555)

### Fixed

> - Workaround for running conda-build using Klocwork (#566)

Reviewer: ```suggestion
- Implemented a workaround for running conda-build using Klocwork (#566)
```

- 225453f Elena Totmenina  Divide pipelines for Linux and Windows (#582)

> - Fix/usm ndarray suai (#584)

Reviewer: ```suggestion
- Fixed inconsistency in `__sycl_usm_array_interface__` of `usm_ndarray` instance (#584)
```

> - Capsule deleters should service used renamed capsules too (#568)

Reviewer: ```suggestion
- Fixed memory leak: Capsule deleters now free resources for renamed capsules too (#568)
```

- Fixed __version__ test to allow for semantic versioning (#569)
- Improved coverage of _types.pxi (#556)

> - Fix UnboundLocalError when default queue could not be created (#554)

Reviewer: ```suggestion
- Fixed `UnboundLocalError` when default queue could not be created (#554)
```
